File: src/main/java/pl/training/workshop/m7/s11_middleman/start/SeatBadge.java
package pl.training.workshop.m7.s11_middleman.start;

import java.util.Objects;

/** Klient 1: plakietka z liczbą wolnych miejsc. Korzysta z pośrednika. */
public final class SeatBadge {
    private final CinemaFacade cinema;

    public SeatBadge(CinemaFacade cinema) {
        this.cinema = Objects.requireNonNull(cinema, "cinema");
    }

    public String badge(String id) {
        int free = cinema.freeSeats(id);
        if (free == 0) {
            return id + ": WYPRZEDANE";
        }
        return cinema.title(id) + " (" + cinema.format(id) + "): " + free + " wolnych";
    }
}

File: typescript/src/workshop/m7/s11_middleman/start/SeatBadge.ts
import { requireNonNull } from '../../../../shared/requireNonNull.js';
import type { CinemaFacade } from './CinemaFacade.js';

/** Klient 1: plakietka z liczbą wolnych miejsc. Korzysta z pośrednika. */
export class SeatBadge {
  private readonly cinema: CinemaFacade;

  constructor(cinema: CinemaFacade) {
    this.cinema = requireNonNull(cinema, 'cinema');
  }

  badge(id: string): string {
    const free = this.cinema.freeSeats(id);
    if (free === 0) {
      return `${id}: WYPRZEDANE`;
    }
    return `${this.cinema.title(id)} (${this.cinema.format(id)}): ${free} wolnych`;
  }
}
